Mathlib/Algebra/Star/NonUnitalSubsemiring.lean

```diff
@@ -0,0 +1,233 @@
+/-
+Copyright (c) 2024 Christopher Hoskin. All rights reserved.
+Released under Apache 2. license as described in the file LICENSE.
+Authors: Christopher Hoskin
+-/
+import Mathlib.Algebra.Ring.Defs
+import Mathlib.Algebra.Group.Subsemigroup.Basic
+import Mathlib.RingTheory.NonUnitalSubsemiring.Basic
+import Mathlib.Algebra.Ring.Subsemiring.Basic
+import Mathlib.Algebra.Star.Center
+
+/-!
+# Non-unital Star Subsemirings
+
+In this file we define `NonUnitalStarSubsemiring`s and the usual operations on them.
+
+## Implementation
+
+This file is heavily inspired by `Mathlib.Algebra.Star.NonUnitalSubalgebra`.
+
+-/
+
+universe v w w'
+
+variable {A : Type v} {B : Type w} {C : Type w'}
+
+/-- A non-unital star subset is a subset which is closed under the `star`
+operation. -/
+structure StarSubset (A : Type v) [Star A] : Type v where
```

Comment:
I'm dubious that we want this. Do you have applications in mind for sets that are closed *only* under the `star` operation?

Reply:
No application in mind at present. Originally I was thinking there might be some efficiency in having a common base class, but I'm not convinced that's actually the case, so I have removed it now.